File: stackflow-frontend/src/app/models/attendance.ts
export type AttendanceStatus = 'Present' | 'Absent' | 'Late' | 'HalfDay' | 'OnLeave';

export interface Attendance {
  id: number;
  employeeId: number;
  employeeName: string;
  date: string;
  status: string;
}

export interface AttendanceRequest {
  employeeId: number;
  date: string;
  status: AttendanceStatus;
}

File: stackflow-api/Models/Attendance.cs
namespace stackflow_api.Models;

public class Attendance
{
    public int Id { get; set; }
    public int EmployeeId { get; set; }
    public Employee Employee { get; set; } = null!;
    public DateOnly Date { get; set; }
    public AttendanceStatus Status { get; set; }
}
